docs/metasploit-framework.wiki/Navigating-and-Understanding-Metasploits-Codebase.md

```diff
@@ -26,22 +26,130 @@ A great outline of Metasploit's code structure can be found at <https://www.offe
+also in turn install `solargraph` itself. If you just want SolarGraph without the Rails integrations, run `gem install solargraph`.
+
+The configuration file for SolarGraph itself can be found at `.solargraph.yml` within the root directory of Metasploit Framework.
+For more information on how this works and how to tweak it, please refer to <https://solargraph.org/guides/configuration>.
+
+Once the Gem files have been installed, the next step is to install the VSCode plugin. You can grab it from 
+<https://marketplace.visualstudio.com/items?itemName=castwide.solargraph>. Once this is done, run the following commands
+to ensure that SolarGraph is using the most up to date information about your code:
+
+```
+bundle install # Update all the gems
+yard gems # Create documentation files for all the gems. SolarGraph relies on YARD for a lot of info.
+yard doc -c # Create YARD docs for all files and use the cache so we don't repeat work (-c option).
+solargraph bundle # Update Solargraph documentation for bundled gems
+```
+
+Then close down VSCode and restart it again, opening up the `metasploit-framework` directory again as a project if needs be.
+This should result in the SolarGraph server starting and then taking a few minutes to index your files. Note that this
+process may occur every time you open up the `metasploit-framework` project. This is normal and to be expected.
+
+If you'd like to save yourself some time, you can have YARD automatically generate new documentation for installed Gems
+by running `yard config --gem-install-yri` which will configure YARD to automatically generate documentation whenever
+new Gems are installed.
+
+# Debugging Metasploit
+
+## Pry Debugging
+Occasionally, simply reading through Metasploit code may not be helpful. You need to actually get into the weeds and learn
+what a piece of code is doing. In these cases, it may be helpful to use `pry`, a Ruby Debugger that can be launched at
+a specific place within your code and which allows you to view the state of the program at that time,
+make adjustments as needed, and then either step through the program or continue to let it run.
+
+You can enter into an interactive debugging environment using `pry` by adding the following code
+snippet within your Metasploit module or library method:
+
+```ruby
+require 'pry'; binding.pry
+```
+
+Pry includes inbuilt commands for code navigation:
+- `backtrace`: Show the current call stack
+- `up` / `down`: Navigate the call stack
+- `step`: Move forward by a single execution step
+- `next`: Move forward by a single line
+- `whereami`: Show the current breakpoint location again
+- `help`: View all of the available commands and options
+  Ruby's runtime introspection can be used to view the available methods, classes, and variables within the current Ruby environment:
+- `self`: To find out what the current object is
```

Comment:
```suggestion
- `help`: View all of the available commands and options

Ruby's runtime introspection can be used to view the available methods, classes, and variables within the current Ruby environment:

- `self`: To find out what the current object is
```

Reply:
Suggested change implemented in https://github.com/rapid7/metasploit-framework/pull/18058/commits/ffa77418c38539e5926558559d1619ca1a079674